src/common/snippets/src/lowered/pass/set_dynamic_wa_to_outermost_loop.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/set_dynamic_wa_to_outermost_loop.hpp"
+
+#include  "snippets/lowered/pass/mha_parallel_wa_optimizer.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/op/brgemm.hpp"
+#include "snippets/utils/loop_utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+bool SetDynamicWAToOuterMostLoop::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SetDynamicWAToOuterMostLoop")
+    if (linear_ir.empty() || !linear_ir.is_dynamic() || linear_ir.get_config().m_enable_domain_optimization)
+        return false;
+
+    const auto linear_ir_ptr = std::make_shared<const LinearIR>(linear_ir);
+    const auto brgemms = MHAParallelWAOptimizer::find_applicable_brgemms(linear_ir_ptr, false);
+    if (brgemms.empty())
+        return false;
+
+    const auto unsqueezed_params = MHAParallelWAOptimizer::find_unsqueezed_params(linear_ir_ptr, brgemms);
+    OPENVINO_ASSERT(!unsqueezed_params.empty(), "unsqueezed_params mustn't be empty after initialization");
+
+
+    const auto& loop_manager = linear_ir_ptr->get_loop_manager();
+    std::unordered_set<lowered::UnifiedLoopInfoPtr> affected_loops;
+    std::vector<size_t> prev_loop_idces;
+    static const size_t dim_M_idx = 1;
+
+    auto add_affected_loop = [&](const lowered::ExpressionPtr& expr) {
+        const auto& loop_idces = expr->get_loop_ids();
+        if (loop_idces != prev_loop_idces) {
```

Comment:
Do we need to compare all `prev_loop_idces` here? We'll work only with `loop_idces.front()` anyway. Should we check only the outermost loop ids here as well?

Reply:
Oh, good catch. Thanks! Missed this vector during copy-pasting.